### DISCUSSION TITLE
gNB upgrade broke 5G UE Connection

### DISCUSSION THREAD

**Discussion by jblakley** [magma/magma#15498] [General]
We upgraded our gNB software and it broke the UE--gNB--AGW transaction. gNB sends the InitialUEMessage; the AGW can't decode that message. After 5 seconds the gNB issues an NGReset. The old version worked OK so expect something changed in the InitialUEMessage that AGW doesn't understand. Can you take a look at this pcap printout and see if anything looks obvious. Suspect something with the changed security information.

**Broken InitialUEMessage.**
[p14111-gnbngresets3-08-21-2024.pdf](https://github.com/user-attachments/files/16714859/p14111-gnbngresets3-08-21-2024.pdf)
[mme_08_21_24.log](https://github.com/user-attachments/files/16714870/mme_08_21_24.log)

**Working InitialUEMessage from the old version.**
[p5-ptc-good-UE-connect.narrow.pdf](https://github.com/user-attachments/files/16714866/p5-ptc-good-UE-connect.narrow.pdf)

**pcap files of the above.**
BAD: [gnbngresets.narrow.zip](https://github.com/user-attachments/files/16714883/gnbngresets.narrow.zip)
GOOD: [ptc-good-UE-connect.narrow.zip](https://github.com/user-attachments/files/16714882/ptc-good-UE-connect.narrow.zip)
@brunohcfaria @jordanvrtanoski @panyogesh @mickymkumar

**Comment by jordanvrtanoski** [comment]
Which release are you using?

**Reply by jblakley** [reply]
Commit f7b296b from 2/26/2024.


Somewhat arbitrary. It was a build for which 5g worked on both sercomm and, until now, gxc.

Is there a better one?

**Comment by jblakley** [comment]
@panyogesh It's been suggested this might be related to the ASN.1 upgrade issue. Is that possible? It is definitely related to AMF decode of the InitialUEMessage. 
```
001484 Wed Aug 21 17:20:52 2024 7F6296FFD700 ERROR NGAP   tasks/ngap/ngap_amf_decoder.c   :0043    Failed to decode PDU
001485 Wed Aug 21 17:20:52 2024 7F6296FFD700 ERROR NGAP   tasks/ngap/ngap_amf.c           :0111    Failed to decode new buffer
```

**Comment by panyogesh** [comment]
@jblakley : Can you please try : https://github.com/magma/magma/pull/15369
If all good we can even think of merging it on 1.9

**Reply by jblakley** [reply]
@panyogesh I tried the patch on master rather than v1.9. Didn't help. We're currently investigating whether this is the issue:
https://github.com/magma/magma/pull/14125/files
It doesn't appear to be in master or v1.9.

**Reply by jblakley** [reply]
Adding this fix on top of v1.9 seems to have fixed the issue. The PDU session is setup and we have end-to-end connectivity between UE and internet.
https://github.com/magma/magma/pull/14125/files
The ASN.1 upgrade was not needed.
@jordanvrtanoski @panyogesh @brunohcfaria

**Comment by jblakley** [comment]
This issue is resolved but awaiting a pull of the fix into v1.9.

In the meantime, these containers on dockerhub include the fix.

```
jblake1/agw_gateway_c:v1.9-asn-file-replace
jblake1/agw_gateway_python:v1.9-asn-file-replace
```

**Comment by jblakley** [comment]
The latest commit to branch v1.9 (384ee01) fixes the issue that we had connecting UE to our gNB. It was this fix discussed above: https://github.com/magma/magma/pull/14125/files. I have pushed two containers for this build  into dockerhub. 
```
jblake1/agw_gateway_c:v1.9
jblake1/agw_gateway_python:v1.9
```
